### PR TITLE
Coral plugin Auth

### PR DESCRIPTION
**Pull request by okbel**
## What does this PR do?
 - Abstracts Auth flow as a plugin `coral-plugin-auth`
 - Adds `embed`, `stream`, `login` slots

## Testing this PR
- Check that every auth path works as expected:
  - Login Flow
  - Sign Up Flow
  - Sign In Flow
  - Forgot Password
  - Change Username, etc.

### CODE REVIEW

**Review by kgardnr — APPROVED** [review]

File: client/coral-admin/src/actions/auth.js [hunk 1]
```diff
@@ -52,7 +52,9 @@ const forgotPassowordFailure = () => ({
 
 export const requestPasswordReset = (email) => (dispatch) => {
   dispatch(forgotPassowordRequest(email));
-  return coralApi('/account/password/reset', {method: 'POST', body: {email}})
+  const redirectUri = location.href;
```
Comment (riley) [inline client/coral-admin/src/actions/auth.js:55]: I think the `location.href` would be the url of the iframe here.

File: plugins/coral-plugin-auth/client/components/SignInButton.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import {Button} from 'coral-ui';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import {showSignInDialog} from 'coral-framework/actions/auth';
+
+const SignInButton = ({loggedIn, showSignInDialog}) => (
+  <div>
+    {!loggedIn
+      ? <Button id="coralSignInButton" onClick={showSignInDialog} full>
+          Sign in to comment
```
Comment (riley) [inline plugins/coral-plugin-auth/client/components/SignInButton.js:11]: this should probably be translated while we're in here

File: plugins/coral-plugin-auth/client/components/SignUpContent.js [hunk 3]
```diff
@@ -1,39 +1,21 @@
...
 
-  constructor (props) {
-    super(props);
-    this.successfulSignup = false;
+    this.state = {
+      successfulSignup: false
+    };
   }
 
-  static propTypes = {
```
Comment (riley) [inline plugins/coral-plugin-auth/client/components/SignUpContent.js:15]: why did we delete all the propTypes?

**Review by jde — APPROVED** [review]

File: client/coral-admin/src/actions/auth.js (hunk 1 above)
Comment (okbel) [reply client/coral-admin/src/actions/auth.js:55]: It’s fine. We need the host there and the popup it’s a permitted host.

File: plugins/coral-plugin-auth/client/components/SignUpContent.js (hunk 3 above)
Comment (okbel) [inline plugins/coral-plugin-auth/client/components/SignUpContent.js]: Restoring them

File: client/coral-admin/src/actions/auth.js (hunk 1 above)
Comment (riley) [reply client/coral-admin/src/actions/auth.js:55]: 👍

**Pull request merged**